async-nats/examples/jetstream_object_store_read_write_delete.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright 2020-2022 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::env;
+use std::str;
+
+use tokio::io::AsyncReadExt;
+
+use async_nats::jetstream;
+
+#[tokio::main]
+async fn main() -> Result<(), async_nats::Error> {
+    // Use the NATS_URL env variable if defined, otherwise fallback to the default.
+    let nats_url = env::var("NATS_URL").unwrap_or_else(|_| "nats://localhost:4222".to_string());
+
+    // Create an unauthenticated connection to NATS.
+    let client = async_nats::connect(nats_url).await?;
+
+    // Access the JetStream Context for managing streams and consumers as well as for publishing and subscription convenience methods.
+    let jetstream = jetstream::new(client);
+
+    // Create a new object bucket
+    let bucket = jetstream
+        .create_object_store(async_nats::jetstream::object_store::Config {
+            bucket: "example_bucket".to_string(),
+            ..Default::default()
+        })
+        .await
+        .unwrap();
+
+    // Create something to store in the bucket. In this case the nats logo.
+    let nats_logo = r#"                                                       
+ /////////////////// ((((((((((((((((((( ################### *******************
+  ///     /////   /// (((((((    (((((((( ###             ### *****         *****
+  ///  */   ///   /// (((((   (((  (((((( ########  ######### ****         ******
+  ///  *////      /// ((((           (((( ########  ######### **** ********  ****
+  /////////////////// ((((((((((((((((((( ################### *******************
+  /////////////////// ((((((((((((((((((( ################### *******************
+             //
+"#;
+
+    // Post the bytes of the nats logo to the bucket.
+    bucket.put("LOGO", &mut nats_logo.as_bytes()).await.unwrap();
```

Comment:
I would prefer having some dummy file to be used, so we can show utilisation of `AsyncRead` trait.
You could just put that ASCI in such file.

Reply:
Sounds good